src/protocols/echo/EchoClient.cpp

```diff
@@ -71,7 +72,7 @@ CHIP_ERROR EchoClient::SendEchoRequest(System::PacketBufferHandle payload)
     }
 
     // Send an Echo Request message.  Discard the exchange context if the send fails.
-    err = mExchangeCtx->SendMessage(kProtocol_Echo, kEchoMessageType_EchoRequest, std::move(payload),
+    err = mExchangeCtx->SendMessage(kProtocol_Echo, static_cast<uint8_t>(MsgType::EchoRequest), std::move(payload),
```

Comment:
this looks worse. Maybe we should make MsgType a namespace instead and have constants like before (or like `constexpr uint8_t` values. static cast is ugly.

Reply:
if `SendMessage()` took a `enum class MsgType` parameter instead of a uint8_t and all the protocols subclassed `MsgType`?

something like:
```
namespace Echo {
enum class MsgType: Exchange::MsgType
```
?